core/src/main/java/org/elasticsearch/cluster/routing/AllocationId.java

```diff
@@ -148,12 +156,52 @@ public String toString() {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject("allocation_id");
-        builder.field("id", id);
+        builder.startObject(ALLOCATION_ID_OBJECT_KEY);
+        builder.field(ID_KEY, id);
         if (relocationId != null) {
-            builder.field("relocation_id", relocationId);
+            builder.field(RELOCATION_ID_KEY, relocationId);
         }
         builder.endObject();
         return builder;
     }
+
+    public static AllocationId fromXContent(XContentParser parser) throws IOException {
```

Comment:
sorry I meant `org.elasticsearch.common.xcontent.ObjectParser` all the time my fault